### PR TITLE
Feature/dark mode

### PR DESCRIPTION
**Pull request by dmouse**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | api-platform/docs

This PR enables the option to change the theme to dark mode, I replace all the colors using css variables and installed a gatsby plugin to manage the theme https://www.gatsbyjs.org/packages/gatsby-plugin-dark-mode 

image
image
image

### CODE REVIEW

File: src/components/layout/Footer.js [hunk 1]
```diff
@@ -49,6 +50,9 @@ class Footer extends Component {
         <p className="footer__policy">
           <Link to="/trademark-policy">Trademark policy</Link>
         </p>
+        <div className="switch">
```
Comment (alanpoulain) [inline src/components/layout/Footer.js:53]:
```suggestion
        <div className="footer__switch">
```

File: src/components/layout/Footer.js [hunk 2]
```diff
@@ -49,6 +50,9 @@ class Footer extends Component {
...
+          <SwitchTheme id="site-menu__dark" />
```
Comment (alanpoulain) [inline src/components/layout/Footer.js:54]: Is this `id` necessary?

File: src/components/common/SwitchTheme.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+const SwitchTheme = () => {
+  return (
+    <ThemeToggler>
+      {({ theme, toggleTheme }) => (
+        <label htmlFor="switch-theme" className="label-switch-theme">
+          <input
+            id="switch-theme"
+            name="switch-theme"
+            className="switch-theme"
```
Comment (alanpoulain) [inline src/components/common/SwitchTheme.js:12]: We are following the BEM (http://getbem.com/) methodology for the CSS.

**Comment by alanpoulain** [comment]
Thank you for this PR :slightly_smiling_face: Seems nice!

**Comment by Deuchnord** [comment]
There is a `box-shadow: white 0px 0px 0px 400px inset` property on the img element of the [distribution](https://api-platform.com/docs/distribution/) homepage, it would be great if you found a way to get rid of it :smiley:

**Comment by alanpoulain** [comment]
Thank you @dmouse!

**Pull request merged**